### ISSUE TITLE
sunAngle() can fail with new R

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2000]
Below is an error message I get from `sunAngle()` given vector arguments.
```R
Error in is.na(az) || any(az < 0) : 
  'length = 1440' in coercion to 'logical(1)'
```
This is because `is.na(x)` is now set up to give an error if `length(x)>1`.

Actually when I say "is" I mean the upcoming R, that is, R-devel.

This is a simple fix but there's no rush, since this is a developer report, and we had a release very recently.

My plan is to fix this and similar in `R/sun.R` and `R/moon.R` on the weekend.  (It's a trivial fix and I've done it already but not pushed to GH.) I will also see if I can figure out a way to detect this problem more generally, with `grep`.  It didn't come up in tests because I was only testing on scalar input ... so I will also add a vector-input tests.

PS. this occurs only if the `time` argument has length exceeding 1.  This was never documented as a possible condition, but I do think it is worth permitting (and documenting) and that's why I will be making code changes.

**Comment by dankelley** [comment]
Fixed in "develop" commit 8d90ac6ee3502e01aaffccb03e51a9de736db7e9.  I rewrote some other aspects that seemed odd to me (e.g. looking for out-of-range values is odd; why not just return what the algorithm gives?).  Luckily, we have a test suite and that still works, plus the application that spurred this (which plots sun and moon traces through the sky) is now working again.

**Comment by richardsc** [comment]
Issue 2000! 😊
